mixins/monorepo.mixin.tf

```
@@ -0,0 +1,63 @@
+# <-- BEGIN DOC -->
+#
+# This mixin is meant to be placed in a Terraform config outside the organization's infrastructure monorepo in order to:
+#
+# 1. Instantiate an AWS Provider using roles managed by the infrastructure monorepo. Since Cloud Posse's providers.tf pattern
+#    requires an invocation of the `account-map` component’s `iam-roles` submodule, which is not present in a repository
+#    outside of the infrastructure monorepo.
+# 2. Retrieve outputs from a component in the infrastructure monorepo. Since Cloud Posse’s `remote-state` module expects
+#    a `stacks` directory, which will not be present in this repository, the monorepo must be cloned via a `monorepo` module
+#    instantiation.
+#
+# Because the source attribute in the `monorepo` and `remote-state` modules cannot be interpolated and refers to a monorepo
+# in a given organization, the following dummy placeholders have been put in place upstream and need to be replaced accordingly
+# when "dropped into" a Terraform configuration:
+#
+# 1. Infrastructure monorepo: `github.com/ACME/infrastructure`
+# 2. Infrastructure monorepo ref: `0.1.0`
+#
+# <-- END DOC -->
+module "monorepo" {
+  source = "git::https://github.com/ACME/infrastructure.git?ref=0.1.0"
+}
+
+locals {
+  monorepo_local_path     = "${path.module}/.terraform/modules/monorepo"
+  stack_config_local_path = "${local.monorepo_local_path}/stacks"
+}
+
+provider "aws" {
+  region = var.region
+
+  profile = module.iam_roles.profiles_enabled ? coalesce(var.import_profile_name, module.iam_roles.terraform_profile_name) : null
+  dynamic "assume_role" {
+    for_each = module.iam_roles.profiles_enabled ? [] : ["role"]
+    content {
+      role_arn = coalesce(var.import_role_arn, module.iam_roles.terraform_role_arn)
+    }
+  }
+}
+
+module "iam_roles" {
+  # https://www.terraform.io/docs/language/modules/sources.html#modules-in-package-sub-directories
+  source                  = "git::https://github.com/ACME/infrastructure.git//components/terraform/account-map/modules/iam-roles?ref=0.1.0"
```

```suggestion
  source                  = "${local.monorepo_local_path}/components/terraform/account-map/modules/iam-roles"
```

to not repeat the remote URL and tag again ?